client/driver/mock_driver.go

```diff
@@ -286,7 +326,12 @@ func (h *mockDriverHandle) run() {
 	for {
 		select {
 		case <-timer.C:
-			close(h.doneCh)
+			select {
+			case <-h.doneCh:
+				// already closed
+			default:
+				close(h.doneCh)
```

Comment:
I ran into a double close panic on the mock driver when testing, so I guarded all of these closes. Not sure why we had never hit it before...